src/desktop/wayland/layer.rs

```diff
@@ -272,7 +272,27 @@ impl LayerMap {
             );
             trace!("Arranging layers into {:?}", output_rect.size);
 
-            for layer in self.layers.iter() {
+            let exclusive_surfaces = self.layers.iter().filter(|l| {
+                matches!(
+                    with_states(l.wl_surface(), |states| states
+                        .cached_state
+                        .get::<LayerSurfaceCachedState>()
+                        .current()
+                        .exclusive_zone),
+                    ExclusiveZone::Exclusive(_)
+                )
```

Comment:
I'd extract this part into a `let is_exclusive = |layer| ...`, also perhaps this needs further anchor checks as described and done below?

> If the surface is:
> 
>    - not anchored
>    - anchored to only two perpendicular edges (a corner),
>    - anchored to only two parallel edges or anchored to all edges,
> 
> The exclusive value should be treated the same as ExclusiveZone::Neutral.